src/test/java/com/uniqueauction/web/trade/controller/PurchaseControllerTest.java

```diff
@@ -0,0 +1,80 @@
+package com.uniqueauction.web.trade.controller;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.uniqueauction.TestContainerBase;
+import com.uniqueauction.domain.trade.service.PurchaseService;
+import com.uniqueauction.web.trade.request.PurchaseRequest;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestContainerBase
+class PurchaseControllerTest {
+
+	private static final Long COMMON_ID = 1L;
+
+	@Autowired
+	private MockMvc mockMvc;
+
+	ObjectMapper objectMapper = new ObjectMapper();
+
+	@MockBean
+	private PurchaseService purchaseService;
+
+	@Test
+	@DisplayName("구매입찰 정상 테스트 201 반환")
+	void purchaseCreateTest() throws Exception {
+
+		mockMvc.perform(
+				post("/purchase")
+					.contentType(MediaType.APPLICATION_JSON)
+					.accept(MediaType.APPLICATION_JSON)
+					.characterEncoding("UTF-8")
+					.content(objectMapper.writeValueAsString(getPurchaseReq())))
+			.andExpect(status().isCreated());
+	}
```

Comment:
테스트에는 정책이 반영되어 있어야 합니다.
이부분은 api 문서부터 조금 오류가 있습니다.

response의 내용이 비어있습니다.
일반적으로 등록시, 등록한 id를 내려주는것이 일반적입니다.
등로후 등록된 정보를 보여준다던가, 등록된 정보로 다른 작업을 이어서 진행하려는 목적이 포함되어있기 때문입니다.

이점이 api문서에도 반영되어야 하며, test에도, 로직에도 반영되어야합니다.
purchase 뿐 아니라 다른 api도 마찬가지입니다.


https://chivalrous-raft-038.notion.site/Unique-Auction-API-895719481430450bb10a6d1d94bae01a#4061aae9deb245759b1b4caa03a64489
<img width="838" alt="image" src="https://user-images.githubusercontent.com/105277168/202232489-cd440402-5652-4162-830e-9a361660f10b.png">



>In case of a POST that resulted in a creation, use a [HTTP 201 status code](http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.5) and include a [Location header](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.30) that points to the URL of the new resource. Both of those in should be in addition to including the newly created resource representation as the body of the response.

https://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api

Reply:
참고해서 추가하도록하겠습니다!